ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -311,6 +316,27 @@ public void deleteInternalTopics(final String applicationId) {
     }
   }
 
+  @Override
+  public Map<TopicPartition, ListOffsetsResultInfo> listTopicOffsets(
+      final String topicName,
+      final OffsetSpec offsetSpec
+  ) {
+    final TopicDescription topicDescription = describeTopic(topicName);
+    final Map<TopicPartition, OffsetSpec> offsetsRequest = new HashMap<>();
+    for (TopicPartitionInfo tpInfo : topicDescription.partitions()) {
+      final TopicPartition tp = new TopicPartition(topicName, tpInfo.partition());
+      offsetsRequest.put(tp, offsetSpec);
+    }
+    try {
+      return ExecutorUtil.executeWithRetries(
+          () -> adminClient.get().listOffsets(offsetsRequest).all().get(),
+          RetryBehaviour.ON_RETRYABLE);
+    } catch (final Exception e) {
+      throw new KafkaResponseGetFailedException(
+          "Failed to get offsets for Kafka Topic " + topicName, e);
+    }
```

Comment:
I think you need something move like to give a better error message to the user on failure:

```suggestion
   } catch (final TopicAuthorizationException e) {
      final Set<String> topics = partitions.stream()
          .map(TopicPartition::topic)
          .collect(Collectors.toSet());

      throw new KsqlTopicAuthorizationException(
          AclOperation.DESCRIBE, topics);
    } catch (final ExecutionException e) {
      throw new KafkaResponseGetFailedException(
          "Failed to get topic offsets. partitions: " + partitions, e.getCause());
    } catch (final Exception e) {
      throw new KafkaResponseGetFailedException(
          "Failed to get topic offsets. partitions: " + partitions, e);
    }
```